cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1314,17 +1315,29 @@ public Secret generateBrokersSecret() {
     /**
      * Generate the Secret containing the username and password to secure the jmx port on the kafka brokers
      *
+     * @param jmxSecretAnnotations The additional annotations of the {@code Secret} created.
+     * @param jmxSecretLabels The additional labels of the {@code Secret} created.
      * @return The generated Secret
      */
-    public Secret generateJmxSecret() {
+    public Secret generateJmxSecret(Map<String, String> jmxSecretAnnotations, Map<String, String> jmxSecretLabels) {
         Map<String, String> data = new HashMap<>(2);
         String[] keys = {SECRET_JMX_USERNAME_KEY, SECRET_JMX_PASSWORD_KEY};
         PasswordGenerator passwordGenerator = new PasswordGenerator(16);
         for (String key : keys) {
             data.put(key, Base64.getEncoder().encodeToString(passwordGenerator.generate().getBytes(StandardCharsets.US_ASCII)));
         }
 
-        return createSecret(KafkaCluster.jmxSecretName(cluster), data);
+        return new SecretBuilder()
+                .withNewMetadata()
+                .withName(KafkaCluster.jmxSecretName(cluster))
+                .withOwnerReferences(createOwnerReference())
+                .withNamespace(namespace)
+                .withLabels(Util.mergeLabelsOrAnnotations(labels.toMap(), jmxSecretLabels))
+                .withAnnotations(jmxSecretAnnotations)
+                .endMetadata()
+                .withType("Opaque")
+                .withData(data)
+                .build();
```

Comment:
Wouldn't it be possible to extend the `createSecret` method with some additional optional parameters? This duplicates a code which will be hard to maintain.